storage/range_raftstorage.go

```diff
@@ -247,6 +247,16 @@ func (r *Range) Snapshot() (raftpb.Snapshot, error) {
 		return raftpb.Snapshot{}, util.Errorf("couldn't find range descriptor")
 	}
 
+	// Store RangeDescriptor in KV[0] as a metadata, it will be retrieved by ApplySnapshot()
+	descValue, err := gogoproto.Marshal(&desc)
+	if err != nil {
+		return raftpb.Snapshot{}, util.Errorf("failed to marshal desc: %s", err)
+	}
+	snapData.KV = append(snapData.KV, &proto.RaftSnapshotData_KeyValue{
```

Comment:
Instead of putting this in `KV[0]`, we should just have a RangeDescriptor field in RaftSnapshotData. That way we don't need to make up a key that will never actually be written to disk.